### PR TITLE
feat(CountDown):Addition of diff props to correct the local time

### PR DESCRIPTION
**Pull request by liixing**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/41945

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      add diff props to  correct the local tiome   |
| 🇨🇳 Chinese |    新增 diff props, 修正本地时间误差      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 82f6fd5</samp>

Added a new feature and improved testing for the `Countdown` component. The new `diff` prop allows users to adjust the time difference between the local and the server. The test case for the `diff` prop uses the `act` function to ensure consistent results.

### <samp>🤖 Generated by Copilot at 82f6fd5</samp>

* Add `diff` prop to `Countdown` component to support custom time difference between local and server time ([link](https://github.com/ant-design/ant-design/pull/41918/files?diff=unified&w=0#diff-fa1fd611b420a3db9c0bcd4b48c78332baa872c5b4d6b4af9253ce7713493cdaR16), [link](https://github.com/ant-design/ant-design/pull/41918/files?diff=unified&w=0#diff-fa1fd611b420a3db9c0bcd4b48c78332baa872c5b4d6b4af9253ce7713493cdaL22-R28), [link](https://github.com/ant-design/ant-design/pull/41918/files?diff=unified&w=0#diff-fa1fd611b420a3db9c0bcd4b48c78332baa872c5b4d6b4af9253ce7713493cdaL39-R48))
* Import `act` function from `../../../tests/utils` to wrap timer updates in test cases ([link](https://github.com/ant-design/ant-design/pull/41918/files?diff=unified&w=0#diff-efc64b56393dbe2b0388a8f3da7f1646518512ed61287ba07cf044da1d7e3a77L7-R7))
* Add test case for `Countdown` component with `diff` prop and check snapshots and callbacks ([link](https://github.com/ant-design/ant-design/pull/41918/files?diff=unified&w=0#diff-efc64b56393dbe2b0388a8f3da7f1646518512ed61287ba07cf044da1d7e3a77R181-R209))
* Swap order of imported types `valueType` and `FormatConfig` in `Countdown.tsx` to match export order in `./utils` ([link](https://github.com/ant-design/ant-design/pull/41918/files?diff=unified&w=0#diff-fa1fd611b420a3db9c0bcd4b48c78332baa872c5b4d6b4af9253ce7713493cdaL6-R6))

### CODE REVIEW

**Comment by Yuiai01** [comment]
新功能的话要提到 feature 分支哈～

**Comment by liixing** [comment]
@Yuiai01  这个我觉得算是 bugfix ，因为现在线上使用倒计时组件有问题。只不过解决方法是新增了 porps

**Comment by li-jia-nan** [comment]
如果是 bug 的话，有办法提供一个复现链接吗？然后看看能不能修复，而不是通过加 featur 去弥补 bug

**Comment by liixing** [comment]
@li-jia-nan  只要本地时间与实际时间对不上就可以复现，因为这个组件就是根据 Date.now() 计算的倒计时

File: components/statistic/__tests__/index.test.tsx [hunk 1]
```diff
@@ -178,6 +178,35 @@ describe('Statistic', () => {
...
+        expect(onFinish).not.toHaveBeenCalled();
+        expect(onChange).toHaveBeenCalled();
+        act(() => {
+          jest.advanceTimersByTime(5000);
+        });
+        expect(container.firstChild).toMatchSnapshot();
+        expect(onFinish).toHaveBeenCalled();
+        expect(onChange).toHaveBeenCalled();
+      });
+    });
```
Comment (Wxh16144) [inline components/statistic/__tests__/index.test.tsx:209]: 倒计时这里用快照测试感觉不太合适 🤔

**Review by Wxh16144 — COMMENTED** [review]
我能理解你描述的问题, 但是不一定要加一个 props 来解, 可以和服务端协商, 请求时携带客户端本地时间, 由服务端返回一些提示 🤔, 不过还是等需要团队核心成员确定一下吧.

**Comment by liixing** [comment]
@Wxh16144  是的，我刚想了一下，也可以传 diff 过后的 value，感谢提醒

**Comment by afc163** [comment]
value 直接减去 diff 就行了，看上去没必要单开 property。

**Pull request closed without merging**